File: ios/Reguerta/Reguerta/Domain/Notifications/ShiftNotificationDetail.swift
import Foundation

struct ShiftNotificationDetail: Equatable {
    let eventID: String
    let assignmentRevision: Int64
    let documentRevision: Int64
    let shift: ShiftAssignment
}

protocol ShiftNotificationDetailRepository: Sendable {
    @MainActor func currentDetail(
        eventID: String,
        memberID: String,
        environment: SessionEnvironment
    ) async throws -> ShiftNotificationDetail
}

struct UnavailableShiftNotificationDetailRepository: ShiftNotificationDetailRepository {
    @MainActor func currentDetail(
        eventID _: String,
        memberID _: String,
        environment _: SessionEnvironment
    ) async throws -> ShiftNotificationDetail {
        throw RepositoryError.unavailable(resource: "notifications.shiftDetail")
    }
}

File: android/Reguerta/app/src/main/java/com/reguerta/user/domain/notifications/ShiftNotificationDetail.kt
package com.reguerta.user.domain.notifications

import com.reguerta.user.domain.shifts.ShiftAssignment

data class ShiftNotificationDetail(
    val eventId: String,
    val assignmentRevision: Long,
    val documentRevision: Long,
    val shift: ShiftAssignment,
)

interface ShiftNotificationDetailRepository {
    suspend fun getCurrentDetail(
        eventId: String,
        memberId: String,
    ): ShiftNotificationDetail
}
